src/coreclr/vm/methodtablebuilder.cpp

```diff
@@ -4428,27 +4428,16 @@ VOID    MethodTableBuilder::InitializeFieldDescs(FieldDesc *pFieldDescList,
                 // The PE should be loaded by now.
                 _ASSERT(GetModule()->GetPEAssembly()->IsLoaded());
 
-                DWORD fldSize;
-                if (FieldDescElementType == ELEMENT_TYPE_VALUETYPE)
-                {
-                    if (IsSelfRef(pByValueClass))
-                    {
-                        _ASSERTE(bmtFP->fHasSelfReferencingStaticValueTypeField_WithRVA);
-
-                        // We do not known the size yet
-                        _ASSERTE(bmtFP->NumInstanceFieldBytes == 0);
-                        // We will check just the RVA with size 0 now, the full size verification will happen in code:VerifySelfReferencingStaticValueTypeFields_WithRVA
-                        fldSize = 0;
-                    }
-                    else
-                    {
-                        fldSize = pByValueClass->GetNumInstanceFieldBytes();
-                    }
-                }
-                else
+#ifdef FEATURE_METADATA_UPDATER
+                // This is a special case for EnC. The RVA field is not actually in the image, but
+                // is instead registered in a dynamic map. We need to set the RVA to a special
+                // value so when the address is looked up, it will be found in the dynamic map.
+                if (GetModule()->IsEditAndContinueEnabled())
```

Comment:
This check does not hurt, but it should not be really needed for perf. `GetDynamicRvaField` has an early out that will be taken for non-EnC modules.

Reply:
I kept this so it is clear this is associated with EnC. Perhaps that isn't interesting here, but it was about being clear when this is involved.